src/components/accessibility/skip_link/skip_link.tsx

```diff
@@ -91,9 +92,29 @@ export const EuiSkipLink: FunctionComponent<EuiSkipLinkProps> = ({
         const destinationEl = document.getElementById(destinationId);
         if (!destinationEl) return;
 
-        destinationEl.scrollIntoView();
-        destinationEl.tabIndex = -1; // Ensure the destination content is focusable
-        destinationEl.focus({ preventScroll: true }); // Scrolling is already handled above, and focus's autoscroll behaves oddly around fixed headers
+        // Scroll to the top of the destination content only if it's ~mostly out of view
+        const destinationY = destinationEl.getBoundingClientRect().top;
+        const halfOfViewportHeight = window.innerHeight / 2;
+        if (
+          destinationY >= halfOfViewportHeight ||
+          window.scrollY >= destinationY + halfOfViewportHeight
+        ) {
+          destinationEl.scrollIntoView();
+        }
+
+        // Ensure the destination content is focusable
+        if (!isTabbable(destinationEl)) {
+          destinationEl.tabIndex = -1;
+          destinationEl.addEventListener(
+            'blur',
+            () => {
+              destinationEl.removeAttribute('tabindex');
+            },
```

Comment:
smol syntax tweak (after writing this out twice I'm getting real good at it hahaa)

```suggestion
            () => destinationEl.removeAttribute('tabindex'),
```